src/rdkafka_admin.c

```diff
@@ -4108,6 +4109,291 @@ rd_kafka_ListOffsetsResponse_parse(rd_kafka_op_t *rko_req,
         return RD_KAFKA_RESP_ERR_NO_ERROR;
 }
 
+/**@}*/
+
+/**
+ * @name ElectLeaders
+ * @{
+ *
+ *
+ *
+ *
+ */
+
+
+/**
+ * @brief Creates a new rd_kafka_ElectLeaders_t object with the given
+ *       \p election_type and \p partitions.
+
+*/
+rd_kafka_ElectLeaders_t *
+rd_kafka_ElectLeaders_new(rd_kafka_ElectionType_t election_type,
+                          rd_kafka_topic_partition_list_t *partitions) {
+
+        rd_kafka_ElectLeaders_t *elect_leaders;
+
+        elect_leaders = rd_calloc(1, sizeof(*elect_leaders));
+        elect_leaders->partitions =
+            rd_kafka_topic_partition_list_copy(partitions);
+        elect_leaders->election_type = election_type;
+
+        return elect_leaders;
+}
+
+/**
+ * @brief Returns a copy of rd_kafka_ElectLeaders_t object.
+ */
+rd_kafka_ElectLeaders_t *
+rd_kafka_ElectLeaders_copy(const rd_kafka_ElectLeaders_t *elect_leaders) {
+        return rd_kafka_ElectLeaders_new(elect_leaders->election_type,
+                                         elect_leaders->partitions);
+}
+
+/**
+ * @brief Destroys the rd_kafka_ElectLeaders_t object.
+ */
+void rd_kafka_ElectLeaders_destroy(rd_kafka_ElectLeaders_t *elect_leaders) {
+        rd_kafka_topic_partition_list_destroy(elect_leaders->partitions);
+        rd_free(elect_leaders);
+}
+
+/**
+ * @brief Frees the memory allocated for an rd_kafka_ElectLeaders_t object
+ *        by calling its destroy function.
+ */
+static void rd_kafka_ElectLeaders_free(void *ptr) {
+        rd_kafka_ElectLeaders_destroy(ptr);
+}
+
+/**
+ * @brief Creates a new rd_kafka_ElectLeadersResult_t object with the given
+ *       \p error.
+ */
+static rd_kafka_ElectLeadersResult_t *
+rd_kafka_ElectLeadersResult_new(rd_kafka_resp_err_t err,
+                                rd_list_t *partitions) {
+
+        rd_kafka_ElectLeadersResult_t *result;
+        result = rd_calloc(1, sizeof(*result));
+
+        result->err = err;
+        rd_list_init_copy(&result->partitions, partitions);
+        rd_list_copy_to(&result->partitions, partitions,
+                        rd_kafka_topic_partition_result_copy_opaque, NULL);
+        return result;
+}
+
+/**
+ * @brief Returns the leader election result object from the
+ *       rd_kafka_ElectLeaders_result_t event type.
+ */
+const rd_kafka_ElectLeadersResult_t *
+rd_kafka_ElectLeaders_result(const rd_kafka_ElectLeaders_result_t *result) {
+        return (const rd_kafka_ElectLeadersResult_t *)rd_list_elem(
+            &result->rko_u.admin_result.results, 0);
+}
+
+/**
+ * @brief Returns the error from the rd_kafka_ElectLeadersResult_t object.
+ */
+rd_kafka_resp_err_t
+rd_kafka_ElectLeadersResult_error(const rd_kafka_ElectLeadersResult_t *result) {
```

Comment:
Remove unnecessary comments from this file. Check older documentations of similar functions in the same file